### PR TITLE
Support for pushing code with SmvGit

### PR DESCRIPTION
**Pull request by laneb**
Currently supports authorization through OAuth token, but we can adapt to support other authorization/authentication protocols as needed.

### CODE REVIEW

File: src/main/python/scripts/smvserver.py [hunk 1]
```diff
@@ -213,6 +213,11 @@ def addFile(author, authorEmail, filePath, commitMessage, workDir="."):
     j_smvPyClient = SmvApp.getInstance().j_smvPyClient
     j_smvPyClient.addFile(author, authorEmail, filePath, commitMessage, workDir)
 
+def pushToRemote(workDir=".", authToken=None):
```
Comment (AliTajeldin) [inline src/main/python/scripts/smvserver.py]: function comment

File: src/main/scala/org/tresamigos/smv/git/SmvGit.scala [hunk 2]
```diff
@@ -41,15 +43,34 @@ case class SmvGit(workDir: String = ".") {
         .setAuthor(author, authorEmail)
         .setMessage(commitMessage)
         .call()
+    }
+
+  def addRemote(remoteName: String, remoteUrl: URL): Unit = {
```
Comment (AliTajeldin) [inline src/main/scala/org/tresamigos/smv/git/SmvGit.scala]: yup, function comment.  when should this be called, how?  does the order matter?

File: src/main/scala/org/tresamigos/smv/git/SmvGit.scala [hunk 3]
```diff
@@ -41,15 +43,34 @@ case class SmvGit(workDir: String = ".") {
...
+    val localConfig = withRepo(workDir) { _.getConfig }
+    val remoteConfig = new RemoteConfig(localConfig, remoteName)
+    val uri = new URIish(remoteUrl)
+    remoteConfig.addURI(uri)
+    remoteConfig.update(localConfig)
+    localConfig.save()
   }
 
+  def pushToRemote(remoteName: String = "origin", optAuthToken: Option[String] = None): Unit =
```
Comment (AliTajeldin) [inline src/main/scala/org/tresamigos/smv/git/SmvGit.scala]: same

**Comment by AliTajeldin** [comment]
might want to just close this PR for now while investigating pure python solution.  If we go back to this, then we can reopen and apply code changes requested here.

**Comment by laneb** [comment]
Because external requirements for git support have changed, this PR is no longer necessary. We will probably remove SmvGit altogether. Closing.

**Pull request closed without merging**